### ISSUE TITLE
3.0.0-beta3 throws XML inflating error

### ISSUE THREAD

**Issue by farukcankaya** [intercom/intercom-android#201]
## Version info
  - Intercom for Android version: 3.0.0-beta3
  - Android OS version: 5.0.1
  - Android Device: GT i9500

## Logs
``````
Fatal Exception: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.twentify.bounty/io.intercom.android.sdk.activities.IntercomPostActivity}: android.view.InflateException: Binary XML file line #57: Error inflating class io.intercom.android.sdk.conversation.ComposerInputView
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2693)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2758)
       at android.app.ActivityThread.access$900(ActivityThread.java:177)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1448)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:5942)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
Caused by android.view.InflateException: Binary XML file line #57: Error inflating class io.intercom.android.sdk.conversation.ComposerInputView
       at android.view.LayoutInflater.createView(LayoutInflater.java:640)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:750)
       at android.view.LayoutInflater.rInflate(LayoutInflater.java:813)
       at android.view.LayoutInflater.rInflate(LayoutInflater.java:821)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:511)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:415)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:366)
       at android.support.v7.app.AppCompatDelegateImplV7.setContentView(AppCompatDelegateImplV7.java:276)
       at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:136)
       at io.intercom.android.sdk.activities.IntercomPostActivity.onCreate(IntercomPostActivity.java:62)
       at android.app.Activity.performCreate(Activity.java:6289)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2646)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2758)
       at android.app.ActivityThread.access$900(ActivityThread.java:177)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1448)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:5942)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
Caused by java.lang.reflect.InvocationTargetException
       at java.lang.reflect.Constructor.newInstance(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:288)
       at android.view.LayoutInflater.createView(LayoutInflater.java:614)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:750)
       at android.view.LayoutInflater.rInflate(LayoutInflater.java:813)
       at android.view.LayoutInflater.rInflate(LayoutInflater.java:821)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:511)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:415)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:366)
       at android.support.v7.app.AppCompatDelegateImplV7.setContentView(AppCompatDelegateImplV7.java:276)
       at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:136)
       at io.intercom.android.sdk.activities.IntercomPostActivity.onCreate(IntercomPostActivity.java:62)
       at android.app.Activity.performCreate(Activity.java:6289)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2646)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2758)
       at android.app.ActivityThread.access$900(ActivityThread.java:177)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1448)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:5942)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'io.intercom.android.sdk.identity.AppIdentity io.intercom.android.sdk.Injector.getAppIdentity()' on a null object reference
       at io.intercom.android.sdk.conversation.ComposerInputView.(ComposerInputView.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:288)
       at android.view.LayoutInflater.createView(LayoutInflater.java:614)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:750)
       at android.view.LayoutInflater.rInflate(LayoutInflater.java:813)
       at android.view.LayoutInflater.rInflate(LayoutInflater.java:821)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:511)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:415)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:366)
       at android.support.v7.app.AppCompatDelegateImplV7.setContentView(AppCompatDelegateImplV7.java:276)
       at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:136)
       at io.intercom.android.sdk.activities.IntercomPostActivity.onCreate(IntercomPostActivity.java:62)
       at android.app.Activity.performCreate(Activity.java:6289)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2646)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2758)
       at android.app.ActivityThread.access$900(ActivityThread.java:177)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1448)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:145)
       at android.app.ActivityThread.main(ActivityThread.java:5942)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
``````

**Comment by ronocod** [comment]
Do you know what situation this error happened in? Was the SDK successfully initialized in `Application.onCreate()`?

**Comment by farukcankaya** [comment]
Yeah, I'm sure that there is no initialization error. I don't know how this error happened, I couldn't reproduce it. It came from one of our users and I want to share with you.

**Comment by ronocod** [comment]
Thanks for sharing it with us, we haven't seen this before and it's a very strange error. `AppIdentity` should only be unavailable before initialization, but you should only be able to receive a Post message after initialization. We'll keep an eye on this and let you know if we find the cause.

**Comment by farukcankaya** [comment]
Thank you, I'll also keep an eye on it

**Comment by ronocod** [comment]
We've released version [3.0.0](https://github.com/intercom/intercom-android/releases/tag/3.0.0) which has some fixes around synchronization and initialization. I still haven't been able to reproduce this but those fixes should stop this from happening. If you still see this issue after updating to 3.0.0 please let us know.
